src/Features/CSharp/Portable/Completion/CompletionProviders/SpeculativeTCompletionProvider.cs

```diff
@@ -47,6 +47,11 @@ public override bool IsTriggerCharacter(SourceText text, int characterPosition,
 
         private async Task<bool> ShouldShowSpeculativeTCompletionItemAsync(Document document, int position, CancellationToken cancellationToken)
         {
+            if (!document.SupportsSyntaxTree)
```

Comment:
Hi @CyrusNajmabadi - I'm noticing that these C# code paths are being called for TypeScript ContainedDocuments. This occurs when a .cshtml file has both razor and JS script blocks. The AbstractSymbolCompletionProvider (and DocumentExtensions) iterate over the related or linked documents which include TS documents.  Here is the initial PR where I tried to filter out the TS documents before this C# code was hit - can you think of a better solution?https://github.com/dotnet/roslyn/pull/9973/commits/563d0a47a4389be1e51242093182af729c2a5790